### DISCUSSION TITLE
Add support for RSP files

### DISCUSSION THREAD

**Discussion by mtribiere** [SCons/scons#4344] [Ideas]
It would be nice to have a support for Response files to avoid the "Too long command line" problem. It would be a really desirable feature especially for larger scale projects. 

It could be as basic as: If there is more than X source files in this command, write all of them in a RSP file and feed the @LIB_A.rsp to the command instead. It could be this format:

`LIB_A.rsp`
```
build/src/fileA.c [\r\n]
build/src/fileB.c [\r\n]
build/src/fileC.c [\r\n]
build/src/fileD.c [\r\n]
```

**Comment by mwichmann** [comment]
There is such a feature

https://scons.org/doc/production/HTML/scons-man.html#cv-TEMPFILE

Support for it is a little erratic, though: it automatically enables on Windows with the msvc compiler, but not necessarily in other cases.

**Reply by mtribiere** [reply]
Alright, I did saw this feature but couldn't get it to work so I assumed I missunderstood something.
I'm on a Windows/Cross-Compilation/GCC type of situation, and I simply need to have a response file for my static libraries.

Could you give me a hint on how I should write that?

**Reply by acmorrow** [reply]
@mtribiere - Have a look at what the mongodb build does to enable TEMPFILE for all links: https://github.com/mongodb/mongo/blob/9dc2751661623af8c071f9f7ce9d9676a2f5469a/SConstruct#L2455-L2473

**Reply by mtribiere** [reply]
Thanks I will check that out
